Basic exploration of the cleaned data

In [1]:
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

pd.set_option("display.max_columns", None)
%config Completer.use_jedi = False

Read in Data

In [2]:
train_df = pd.read_csv("../data/train_cleaned.csv")
train_df["data_sample"] = "train"

dev_df = pd.read_csv("../data/dev_cleaned.csv")
dev_df["data_sample"] = "dev"

test_df = pd.read_csv("../data/test_cleaned.csv")
test_df["data_sample"] = "test"

# combine together
all_df = pd.concat([train_df, dev_df, test_df])
all_df.reset_index(inplace=True)
all_df.drop(columns=["index"], inplace=True)

all_df.head()

Payment_Behavior_!@9#%8  Payment_Behavior_High_spent_Large_value_payments  \
0                        0                                                 0   
1                        0                                                 0   
2                        0                                                 0   
3                        0                                                 0   
4                        0                                                 1   

   Payment_Behavior_High_spent_Medium_value_payments  \
0                                                  0   
1                                                  1   
2                                                  0   
3                                                  0   
4                                                  0   

   Payment_Behavior_High_spent_Small_value_payments  \
0                                                 0   
1                                                 0   
2                                                 0   
3                                                 1   
4                                                 0   

   Payment_Behavior_Low_spent_Large_value_payments  \
0                                                0   
1                                                0   
2                                                0   
3                                                0   
4                                                0   

   Payment_Behavior_Low_spent_Medium_value_payments  \
0                                                 0   
1                                                 0   
2                                                 0   
3                                                 0   
4                                                 0   

   Payment_Behavior_Low_spent_Small_value_payments  Credit_Mix_Bad  \
0                                                1               0   
1                                                0               1   
2                                                1               0   
3                                                0               0   
4                                                0               0   

   Credit_Mix_Good  Credit_Mix_Standard  Credit_Mix__  and auto loan  \
0                0                    0             1              1   
1                0                    0             0              1   
2                0                    0             1              0   
3                0                    0             1              0   
4                0                    1             0              0   

   and credit-builder loan  and debt consolidation loan  and home equity loan  \
0                        0                            0                     0   
1                        0                            0                     0   
2                        0                            0                     0   
3                        0                            0                     0   
4                        0                            0                     0   

   and mortgage loan  and not specified  and payday loan  and personal loan  \
0                  0                  0                0                  0   
1                  0                  0                0                  0   
2                  0                  0                0                  0   
3                  0                  0                1                  0   
4                  0                  0                0                  0   

   and student loan  auto loan  credit-builder loan  debt consolidation loan  \
0                 0          0                    0                        0   
1                 0          0                    0                        0   
2                 1          1                    0                        0   
3                 0          0                    0           

Distribution of our labels

In [3]:
# aggregate by "Credit_Score" and "data_sample"
count_by_score_df = (pd.DataFrame(all_df
                                 .groupby(["Credit_Score", "data_sample"])["Credit_Score"]
                                 .count())
                     .rename(columns={"Credit_Score": "count"})
                     .reset_index())


# aggregate by "data_sample": will use values for normalization
count_by_sample_df = (pd.DataFrame(all_df
                                 .groupby(["data_sample"])["Credit_Score"]
                                 .count())
                     .rename(columns={"Credit_Score": "total_sample_count"})
                     .reset_index())


all_agg_df = pd.merge(count_by_score_df, count_by_sample_df, on="data_sample", how="inner")
all_agg_df["perc"] = all_agg_df["count"] / all_agg_df["total_sample_count"]
all_agg_df.sort_values(["Credit_Score", "data_sample"], inplace=True)
all_agg_df

Credit_Score data_sample  count  total_sample_count      perc
0             0         dev    272                1875  0.145067
3             0        test    333                1875  0.177600
6             0       train   1370                8750  0.156571
1             1         dev    527                1875  0.281067
4             1        test    576                1875  0.307200
7             1       train   2479                8750  0.283314
2             2         dev   1076                1875  0.573867
5             2        test    966                1875  0.515200
8             2       train   4901                8750  0.560114


From the table above, we can see that the `Credit_Score` classes are not balanced.

* 0: ~15%
* 1: ~30%
* 2: ~55%


All of the classes are at least on the same order of magntude (in terms of percentage).
So this shouldn't be too bad to work with.